base/int.jl

```diff
@@ -60,7 +60,7 @@ abs(x::Unsigned) = x
 abs(x::Signed) = flipsign(x,x)
 
 """
-    Base.checked_abs(x)
+    Base.checked_abs(x::Integer)
```

Comment:
Rather than changing the signature in the docs, wouldn't it be better to add implementations for floating point that call the standard functions (like `abs()`)? For generic programming, you need these operations to work with all types, just like their unchecked equivalents.